telegram/ext/conversationhandler.py

```diff
@@ -354,12 +354,16 @@ def update_state(self, new_state, key):
             if self.persistent:
                 self.persistence.update_conversation(self.name, key, new_state)
 
-    def _trigger_timeout(self, bot, job):
+    def _trigger_timeout(self, context, job=None):
         self.logger.debug('conversation timeout was triggered!')
-        del self.timeout_jobs[job.context.conversation_key]
+        if isinstance(context, CallbackContext):
```

Comment:
Could you add a comment explaining why we are doing all this mess in the first place? :)

Reply:
Alright! I've done it, do let me know what you think.